src/Generating/ComposableGenerator.cpp

```diff
@@ -438,6 +438,32 @@ void cComposableGenerator::InitFinishGens(cIniFile & a_IniFile)
 				m_FinishGens.push_back(gen);
 			}
 		}
+		else if (NoCaseCompare(finisher, "OverworldClumpFlowers") == 0)
+		{
+			std::vector<cFinishGenClumpTopBlock::BiomeInfo> flowers;
+			int NumGeneratorValues = a_IniFile.GetNumValues("Generator");
+			int GeneratorKeyId = a_IniFile.FindKey("Generator");
+			for (int i = 0; i < NumGeneratorValues; i++)
+			{
+				AString ValueName = a_IniFile.GetValueName("Generator", i);
+				if (ValueName.substr(0, AString("OverworldClumpFlowers-").size()) == "OverworldClumpFlowers-")
+				{
+					AString & RawClump = a_IniFile.GetValue(GeneratorKeyId, i);
+					cFinishGenClumpTopBlock::ParseConfigurationString(RawClump, flowers);
+				}
+			}
+
+			if (flowers.size() == 0)
+			{
+				cFinishGenClumpTopBlock::ParseConfigurationString(a_IniFile.GetValueSet("Generator", "OverworldClumpFlowers-1", "Forest+-2+2,ForestHills+-3+2,FlowerForest; yellowflower,redflower,lilac,rosebush"));
+				cFinishGenClumpTopBlock::ParseConfigurationString(a_IniFile.GetValueSet("Generator", "OverworldClumpFlowers-2", "Plains+-2+1,SunflowerPlains; yellowflower,redflower,azurebluet,oxeyedaisy"));
+				cFinishGenClumpTopBlock::ParseConfigurationString(a_IniFile.GetValueSet("Generator", "OverworldClumpFlowers-3", "SunflowerPlains+1+2; sunflower"));
+				cFinishGenClumpTopBlock::ParseConfigurationString(a_IniFile.GetValueSet("Generator", "OverworldClumpFlowers-4", "FlowerForest+2+5; allium,redtulip,orangetulip,whitetulip,pinktulip,oxeyedaisy"));
+				cFinishGenClumpTopBlock::ParseConfigurationString(a_IniFile.GetValueSet("Generator", "OverworldClumpFlowers-5", "Swampland,SwamplandM; brownmushroom,redmushroom,blueorchid"));
+			}
+			
+			m_FinishGens.push_back(cFinishGenPtr(new cFinishGenClumpTopBlock(Seed, flowers)));
```

Comment:
All the other generators that have non-trivial initialization had used the construct-initialize-use sequence, where the constructor is trivial and initialization reads from the ini file. That way, we can make the `Initialize()` function virtual in the base class and just call it on all generators after they are constructed.

I think the best place for the default values is in the generator implementation itself. If there ever is a need to edit the defaults, it usually means the need to edit the generator anyway, so that it's in one place. Searching for the defaults in a completely different file seems counter-intuitive for me.